VimeoNetworking/Sources/Models/VIMUpload.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  VIMUpload.swift
+//  Pods
+//
+//  Created by Lehrer, Nicole on 4/18/18.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+/// Encapsulates the upload-related information on a video object.
+public class VIMUpload: VIMModelObject
+{
+    /// The approach for uploading the video, expressed as a Swift-only enum
+    ///
+    /// - streaming: Two-step upload without using tus; this will be deprecated
+    /// - post: Upload with an HTML form or POST
+    /// - pull: Upload from a video file that already exists on the internet
+    /// - tus: Upload using the open-source tus protocol
+    public enum UploadApproach: String
```

Comment:
Just curious why do we need to know the upload approach used? Should we use it in the creation of a video request?

Reply:
Since we receive it on the object I thought we could define this enum here but I do like using this value inside upload instead of redefining. Updating it there.